akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala

```diff
@@ -120,6 +119,46 @@ object GraphStages {
     }
   }
 
+  private class TerminationWatcher[T] extends GraphStageWithMaterializedValue[FlowShape[T, T], Future[Unit]] {
+    val name = "TerminationWatcher"
+    val in = Inlet[T](name + ".in")
+    val out = Outlet[T](name + ".out")
+    override def initialAttributes = Attributes.name(name)
```

Comment:
please add to GraphStages.DefaultAttributes so that they are reused—also put the final port names as strings into the file to profit from the class file’s constant pool